app/models/work_package/time_entries_cleaner.rb

```diff
@@ -56,7 +56,7 @@ def cleanup_time_entries_before_destruction_of(work_packages,
     end
 
     def update_time_entries(work_packages, action)
-      TimeEntry.where(["work_package_id IN (?)", work_packages.map(&:id)]).update_all(action)
+      TimeEntry.where(["entity_type = ? AND entity_id IN (?)", "WorkPackage", work_packages.map(&:id)]).update_all(action)
```

Comment:
I think you can just do:

```suggestion
      TimeEntry.where(entity: work_packages).update_all(action)
```

Reply:
For a polymorphic association it is actually better to keep it that way, as your suggestion would generate a query like `(entity_type="WorkPackage" AND entity_id=1) OR (entity_type = "WorkPackage" AND entit_id=2)`. This way the query is more efficient.